lib/ring.gd

```diff
@@ -623,8 +623,6 @@ InstallSubsetMaintenance( IsIntegralRing,
 
 InstallTrueMethod( IsIntegralRing,
     IsRing and IsMagmaWithInversesIfNonzero and IsNonTrivial );
-InstallTrueMethod( IsIntegralRing,
-    IsUniqueFactorizationRing and IsNonTrivial );
```

Comment:
But according to the documentation for `IsUniqueFactorizationRing`, this is correct! If we make this change, we also ought to change the documentation.

But I think the true cause is elsewhere. As I commented on #3975, the commit which introduced this regression is 0f9c10479318a4ae19ac18eb494e22b6fae29527 and it adds another implication which might be the root cause:
```gap
InstallTrueMethod(IsEuclideanRing, IsZmodnZObjNonprimeCollection and
    IsWholeFamily and IsRing);
```
Indeed, I overlooked that the documentation for `IsEuclideanRing` says: 
> A  ring  R  is  called  a  Euclidean  ring  if  it is an integral ring

Oops. So *that* implication is the culprit.

That said, I think it is unfortunate that we define euclidean rings as being integral. So right now I think the quick fix would be to remove the implication I added, but the better fix would be to change the documentation and implications for `IsEuclideanRing`.

I now have a meeting but can look at this later.

Reply:
Oh, no, quick fix does not work: just removing the implication causes this:
```
Error, required filters [ "IsListOrCollection", "IsCollection", "IsFinite", "HasIsFinite", "IsWholeFamily", "HasIsWholeFamily", 
  "IsDuplicateFree", "HasIsDuplicateFree", "IsExtAElement", "CategoryCollections(IsExtAElement)", 
  "CategoryCollections(IsNearAdditiveElement)", "CategoryCollections(IsNearAdditiveElementWithZero)", 
  "CategoryCollections(IsNearAdditiveElementWithInverse)", "CategoryCollections(IsAdditiveElement)", 
  "IsExtLElement", "CategoryCollections(IsExtLElement)", "IsExtRElement", "CategoryCollections(IsExtRElement)", 
  "CategoryCollections(IsMultiplicativeElement)", "CategoryCollections(IsMultiplicativeElementWithOne)", 
  "CategoryCollections(IsMultiplicativeElementWithInverse)", "CategoryCollections(IsAssociativeElement)", 
  "CategoryCollections(IsAdditivelyCommutativeElement)", "CategoryCollections(IsCommutativeElement)", 
  "IsGeneralizedDomain", "IsMagma", "IsAssociative", "HasIsAssociative", "IsCommutative", "HasIsCommutative", 
  "IsGeneratorsOfSemigroup", "HasIsGeneratorsOfSemigroup", "IsNearAdditiveMagma", "IsNearAdditiveMagmaWithZero", 
  "IsNearAdditiveGroup", "IsAdditivelyCommutative", "HasIsAdditivelyCommutative", "IsLDistributive", 
  "HasIsLDistributive", "IsRDistributive", "HasIsRDistributive", "CategoryCollections(IsZmodnZObjNonprime)" ]
for 1st argument do not match a declaration of EuclideanDegree called from
<<compiled GAP code>> from GAPROOT/lib/oper1.g:338 in function InstallMethod called from
```

Reply:
As far as I can see, `IsIntegralRing` is what is usually called integral domain (but domain  has a technical meaning in GAP). I thus do not see the problem with Euclidean rings being defined to be integral, respectively why should Z/6Z be Euclidean?

Reply:
As @fieker explained below, I'd turn the question around: I see now reason why one should restrict "euclidean rings" to be integral, this is arbitrary.